src/Widgets/DataFormBooleanWidget/DataFormBooleanWidgetComponent.tsx

```diff
@@ -10,8 +10,13 @@ import './DataFormBooleanWidget.scss'
 
 provideComponent(DataFormBooleanWidget, ({ widget }) => {
   const attributeName = useData().attributeName()
-  const id = ['DataFormBooleanWidget', widget.id(), attributeName].join('-')
-
+  const elementId = useData().dataItem()?.id()
+  const id = [
+    'DataFormBooleanWidget',
+    widget.id(),
+    elementId,
+    attributeName,
+  ].join('-')
```

Comment:
It's more personal taste, but I would inline `elementId`. This applies to the other places as well.